qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py

```diff
@@ -74,11 +76,120 @@ def get_raw(self) -> np.ndarray:
                                                 self.steps,
                                                 self.mode)
         else:
-            raise RuntimeError("No instrument attached to Parameter")
+            raise RuntimeError("No instrument attached to Parameter.")
 
         return data
 
 
+class TimeTrace(ParameterWithSetpoints):
+    """
+    A parameter class that holds the data corresponding to the time dependence of
+    current and voltage.
+    """
+
+    def __init__(self, name: str, instrument: Instrument, **kwargs) -> None:
+
+        super().__init__(name=name,
+                         instrument=instrument,
+                         **kwargs)
+
+    def _check_time_trace(self) -> None:
+        """
+        A helper function that compares the integration time with measurement
+        interval for accurate results.
+
+        Raises:
+            RuntimeError: If no instrument attached to Parameter.
+        """
+        if self.instrument is None:
```

Comment:
Yes you are right, driver works perfectly fine without them. However, CI did not like the code without them in my early commits, so I had to add these checks.  Thus, they should stay.

Reply:
hm... would it work if this check goes to the `__init__` and `mypy` is made happy in that one place? instead of duplicating this check in all the methods of the class

Reply:
Yes, I concur with your criticism of repeating code lines, however, moving it to ``__init__`` unfortunately won't work. I will keep this conversation open until you are satisfied, however, driver is ready to be merged.